src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -415,6 +438,13 @@ def get_command_table(module_name=None):
         logger.debug("Loaded all modules in %.3f seconds. "
                      "(note: there's always an overhead with the first module loaded)",
                      cumulative_elapsed_time)
+    try:
+        # We always load extensions even if the appropriate module has been loaded
+        # as an extension could override the commands already loaded.
```

Comment:
1. Changing source code on disk would be pretty difficult (or should be).  If you could do it, you've already got access to their PC so you've got free reign anyways.
2. Extensions by their very nature make it technically much easier to get people to run untrusted, malicious code.
3. Allowing extensions to override trusted CLI commands now makes it **even easier** to engineer a social attack because the message isn't "install my sketchy extension" but "install these great improvements to the VM commands". 

I like Troy's idea of a warning when running any overridden commands. I would also suggest considering a strong warning when an extension install would override any CLI commands and a prompt that yes, you want to install this potentially dangerous thing.